diff --git a/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootView.kt b/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootView.kt
index ebce5f9fa39e451ccd4f7f2515913ad5e91a2fc5..da0b4873f4909a8d0e1d14f552863dcb048aa47c 100644
--- a/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootView.kt
+++ b/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootView.kt
@@ -32,17 +32,24 @@ class RNGestureHandlerRootView(context: Context?) : ReactViewGroup(context) {
     rootHelper?.tearDown()
   }
 
-  override fun dispatchTouchEvent(ev: MotionEvent) = if (rootViewEnabled && rootHelper!!.dispatchTouchEvent(ev)) {
-    true
-  } else {
-    super.dispatchTouchEvent(ev)
-  }
+  override fun dispatchTouchEvent(ev: MotionEvent) =
+    try {
+      if (rootViewEnabled && rootHelper!!.dispatchTouchEvent(ev)) {
+        true
+      } else super.dispatchTouchEvent(ev)
+    } catch (e: IllegalArgumentException) {
+      e.printStackTrace()
+      false
+    }
 
   override fun dispatchGenericMotionEvent(event: MotionEvent) =
-    if (rootViewEnabled && rootHelper!!.dispatchTouchEvent(event)) {
-      true
-    } else {
-      super.dispatchGenericMotionEvent(event)
+    try {
+      if (rootViewEnabled && rootHelper!!.dispatchTouchEvent(ev)) {
+        true
+      } else super.dispatchGenericMotionEvent(ev)
+    } catch (e: IllegalArgumentException) {
+      e.printStackTrace()
+      false
     }
 
   override fun requestDisallowInterceptTouchEvent(disallowIntercept: Boolean) {
@@ -79,3 +86,4 @@ class RNGestureHandlerRootView(context: Context?) : ReactViewGroup(context) {
     }
   }
 }
+
