diff --git a/node_modules/@react-native-clipboard/clipboard/android/build.gradle b/node_modules/@react-native-clipboard/clipboard/android/build.gradle
index 86d4866..0943c37 100644
--- a/node_modules/@react-native-clipboard/clipboard/android/build.gradle
+++ b/node_modules/@react-native-clipboard/clipboard/android/build.gradle
@@ -22,6 +22,11 @@ apply plugin: 'com.android.library'
 android {
     compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')

+    def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+    if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+        namespace "com.reactnativecommunity.clipboard"
+    }
+
     defaultConfig {
         minSdkVersion getExtOrIntegerDefault('minSdkVersion')
         targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
