diff --git a/node_modules/react-native-quick-md5/android/build.gradle b/node_modules/react-native-quick-md5/android/build.gradle
index 53a7471..07acabe 100644
--- a/node_modules/react-native-quick-md5/android/build.gradle
+++ b/node_modules/react-native-quick-md5/android/build.gradle
@@ -45,6 +45,11 @@ android {
   buildToolsVersion getExtOrDefault('buildToolsVersion')
   ndkVersion getExtOrDefault('ndkVersion')

+  def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+  if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+    namespace "com.reactnativequickmd5"
+  }
+
   defaultConfig {
     minSdkVersion 21
     targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
