diff --git a/node_modules/react-native-keep-awake/android/build.gradle b/node_modules/react-native-keep-awake/android/build.gradle
index ba4dfe7..2dae65d 100644
--- a/node_modules/react-native-keep-awake/android/build.gradle
+++ b/node_modules/react-native-keep-awake/android/build.gradle
@@ -23,6 +23,11 @@ android {
     compileSdkVersion _compileSdkVersion
     buildToolsVersion _buildToolsVersion

+    def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+    if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+        namespace "com.corbt.keepawake"
+    }
+
     defaultConfig {
         minSdkVersion _minSdkVersion
         targetSdkVersion _targetSdkVersion
