diff --git a/node_modules/react-native-track-player/android/build.gradle b/node_modules/react-native-track-player/android/build.gradle
index ac33473..b98bc26 100644
--- a/node_modules/react-native-track-player/android/build.gradle
+++ b/node_modules/react-native-track-player/android/build.gradle
@@ -18,6 +18,10 @@ def getExtOrIntegerDefault(name) {
 }
 
 android {
+    buildFeatures {
+        buildConfig = true
+    }
+
     compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
     namespace 'com.doublesymmetry.trackplayer'
 
