diff --git a/node_modules/rollbar-react-native/android/build.gradle b/node_modules/rollbar-react-native/android/build.gradle
index e156841..4619317 100644
--- a/node_modules/rollbar-react-native/android/build.gradle
+++ b/node_modules/rollbar-react-native/android/build.gradle
@@ -1,12 +1,21 @@
 apply plugin: 'com.android.library'
 
 android {
+    buildFeatures {
+        buildConfig = true
+    }
+
     compileSdkVersion project.hasProperty('compileSdkVersion') ? project.compileSdkVersion : 27
     buildToolsVersion project.hasProperty('buildToolsVersion') ? project.buildToolsVersion : "27.0.3"
 
+    def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+    if (agpVersion.tokenize('.')[0].toInteger() >= 7) {
+        namespace "com.rollbar"
+    }
+
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 27
+        minSdkVersion project.hasProperty('minSdkVersion') ? project.minSdkVersion : 16
+        targetSdkVersion project.hasProperty('targetSdkVersion') ? project.targetSdkVersion : 27
         versionCode 1
         versionName "1.0"
     }
